src/cgame/cg_view.cpp

```diff
@@ -1330,94 +1330,65 @@ Sets cg.refdef.gradingWeights
 */
 static void CG_CalcColorGradingForPoint( vec3_t loc )
 {
-	int   i, j;
-	float dist, weight;
-	int   selectedIdx[3] = { 0, 0, 0 };
-	float selectedWeight[3] = { 0.0f, 0.0f, 0.0f };
-	float totalWeight = 0.0f;
-	int freeSlot = -1;
-	bool haveGlobal = false;
+	// search 3 greatest weights
+	int selectedTex[ 4 ]{};
+	float selectedWeight[ 4 ]{};
 
-	// the first allocated grading is special in that it may be global
-	i = 0;
-
-	if ( cgs.gameGradingTextures[0] && cgs.gameGradingModels[0] == -1 )
+	for( int i = 0; i < MAX_GRADING_TEXTURES; i++ )
 	{
-		selectedIdx[0] = 0; // shouldn't be needed
-		selectedWeight[0] = 2.0f; // won't be sorted down
-		haveGlobal = true;
-		i = 1;
-	}
-
-	for(; i < MAX_GRADING_TEXTURES; i++ )
-	{
-		if( !cgs.gameGradingTextures[i] )
+		if ( !cgs.gameGradingTextures[i] || cgs.gameGradingModels[i] < 0 )
 		{
 			continue;
 		}
 
-		dist = CM_DistanceToModel( loc, cgs.gameGradingModels[i] );
-		weight = 1.0f - dist / cgs.gameGradingDistances[i];
-		weight = Math::Clamp( weight, 0.0f, 1.0f ); // Maths::clampFraction( weight )
-
-		// search 3 greatest weights
-		if( weight <= selectedWeight[2] )
-		{
-			continue;
-		}
+		float dist = CM_DistanceToModel( loc, cgs.gameGradingModels[i] );
+		float weight = 1.0f - dist / std::max( 0.01f, cgs.gameGradingDistances[i] );
 
-		for( j = 1; j >= 0; j-- )
+		for ( int j = 1; j < 4 && weight > selectedWeight[ j ]; j++ )
 		{
-			if( weight <= selectedWeight[j] )
-			{
-				break;
-			}
-
-			selectedIdx[j+1] = selectedIdx[j];
-			selectedWeight[j+1] = selectedWeight[j];
+			selectedTex[ j - 1 ] = selectedTex[ j ];
```

Comment:
These seem to always be 0 at this point.

Reply:
Do you mean based on inspecting the code or looking at it in a debugger? If the latter, it's because the only known map build using the feature is https://cdn.unvanquished.net/unvanquished_0.24.1_universal/pkg/map-plat23_b11.pk3

Reply:
By looking at the code. I see `selectedTex` having its values set only where it is declared.

Reply:
I see. I have the screenshots from when I (attempted to) test this patch with that map and they are correct... must have made some mistake with building or test commands. I'll take a look later.

Reply:
Added the missing line `selectedTex[ j ] = cgs.gameGradingTextures[ i ];`. Now it's working.